core/templates/domain/topic_viewer/read-only-topic-object.factory.ts

```diff
@@ -0,0 +1,156 @@
+  [skillId: string]: number | null;
+}
+
+export class ReadOnlyTopic {
+  _topicName: string;
+  _topicId: string;
+  _canonicalStorySummaries: Array<StorySummary>;
+  _additionalStorySummaries: Array<StorySummary>;
+  _uncategorizedSkillSummaries: Array<SkillSummary>;
+  _subtopics: Array<Subtopic>;
+  _degreesOfMastery: IDegreesOfMastery;
+  _skillDescriptions: ISkillDescriptions;
+  _trainTabShouldBeDisplayed: boolean;
+
+  constructor(
+      topicName: string, topicId: string,
+      canonicalStorySummaries: Array<StorySummary>,
+      additionalStorySummaries: Array<StorySummary>,
+      uncategorizedSkillSummaries: Array<SkillSummary>,
+      subtopics: Array<Subtopic>,
+      degreesOfMastery: IDegreesOfMastery,
+      skillDescriptions: ISkillDescriptions,
+      trainTabShouldBeDisplayed: boolean) {
+    this._topicName = topicName;
+    this._topicId = topicId;
+    this._canonicalStorySummaries = canonicalStorySummaries;
+    this._additionalStorySummaries = additionalStorySummaries;
+    this._uncategorizedSkillSummaries = uncategorizedSkillSummaries;
+    this._subtopics = subtopics;
+    this._degreesOfMastery = degreesOfMastery;
+    this._skillDescriptions = skillDescriptions;
+    this._trainTabShouldBeDisplayed = trainTabShouldBeDisplayed;
+  }
+
+  getTopicName(): string {
+    return this._topicName;
+  }
+
+  getTopicId(): string {
+    return this._topicId;
+  }
+
+  getCanonicalStorySummaries(): Array<StorySummary> {
+    return this._canonicalStorySummaries.slice();
+  }
+
+  getAdditionalStorySummaries(): Array<StorySummary> {
+    return this._additionalStorySummaries.slice();
+  }
+
+  getUncategorizedSkillsSummaries(): Array<SkillSummary> {
+    return this._uncategorizedSkillSummaries.slice();
+  }
+
+  getSubtopics(): Array<Subtopic> {
+    return this._subtopics.slice();
+  }
+
+  getDegreesOfMastery(): IDegreesOfMastery {
+    return this._degreesOfMastery;
+  }
+
+  getSkillDescriptions(): ISkillDescriptions {
+    return this._skillDescriptions;
+  }
+
+  getTrainTabShouldBeDisplayed(): boolean {
+    return this._trainTabShouldBeDisplayed;
+  }
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ReadOnlyTopicObjectFactory {
+  _subtopicObjectFactory: SubtopicObjectFactory;
+  _skillSummaryObjectFactory: SkillSummaryObjectFactory;
+
+  constructor(
+    private subtopicObjectFactory: SubtopicObjectFactory,
+    private skillSummaryObjectFactory: SkillSummaryObjectFactory) {
+    this._subtopicObjectFactory = subtopicObjectFactory;
+    this._skillSummaryObjectFactory = skillSummaryObjectFactory;
+  }
+  // TODO(#7165): Replace any with exact type
+  createFromBackendDict(topicDataDict: any): ReadOnlyTopic {
+    let subtopics: Array<Subtopic> = topicDataDict.subtopics.map(
+      (subtopic: Subtopic) => {
+        return this._subtopicObjectFactory.create(
+          subtopic, topicDataDict.skill_descriptions);
+      });
+    let uncategorizedSkills: Array<SkillSummary> =
```

Comment:
Wrap inside `(..)` and break after `(` here and below. Please follow the coding style [here](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript).

Reply:
@kevintab95 , are you talking 1st point of [this](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript)

Reply:
@Vinit-Dantkale,  Yes, that's what @kevintab95 wants to highlight, he also wants you to make sure that the code style follows all the guidelines provided there.